src/test/regress/expected/multi_transaction_recovery.out

```diff
@@ -410,7 +410,7 @@ COMMIT;
 SELECT count(*) FROM pg_dist_transaction;
  count
 ---------------------------------------------------------------------
-     2
+     4
```

Comment:
However, this change seems unexpected to me. Should we maybe call `recover_prepared_transactions()` before `only two of the connections will perform a write (INSERT)`? 

We might have the 2 extra entries due to `citus_move_shard_placement` in between